src/Controller/BannerController.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\BannerServer\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use WMDE\BannerServer\UseCase\BannerProvideRequest;
+use WMDE\BannerServer\UseCase\BannerProvideUseCase;
+
+/**
+ * @license GNU GPL v2+
+ */
+class BannerController {
+
+	public function provide(): Response {
+		$useCase = new BannerProvideUseCase();
+		return $useCase->processBannerRequest( $this->buildRequest() );
+	}
+
+	private function buildRequest(): BannerProvideRequest {
+		$request = Request::createFromGlobals();
```

Comment:
Again, creating the request here (for the 2nd time, see index.php), will make the controller less testable than if you pass it in via parameter in `provide`. See https://symfony.com/doc/current/controller.html#the-request-object-as-a-controller-argument